main.go

```diff
@@ -59,22 +67,38 @@ func RunPrompt() {
 			//if we are in a block, match the indentation of above
 			fmt.Print("  ")
 		}
-		in, _ := reader.ReadString('\n')
+		in, err := reader.ReadString('\n')
+		if err == io.EOF {
+			fmt.Printf("\n REPL detected EOF. exiting...")
+			os.Exit(1)
```

Comment:
Exit code 1 indicates error. I think this is normal behavior.

```suggestion
			os.Exit(0)
```

Reply:
Addressed in new commit